alica_test_utility/src/Util.cpp

```diff
@@ -68,6 +68,30 @@ bool Util::isPlanActiveHelper(const RunningPlan* rp, int64_t id)
     return false;
 }
 
+bool Util::hasPlanSucceeded(alica::AlicaEngine* ae, int64_t id)
+{
+    return hasPlanSucceededHelper(ae->getPlanBase().getRootNode(), id);
+}
+
+bool Util::hasPlanSucceededHelper(const RunningPlan* rp, int64_t id)
```

Comment:
I may not be fully clear on success mechanics, but this isn't really checking one specific plan but rather just if any plan at any level is succeeded right?

Reply:
No this method will

- Recursively traverse the runningPlan tree
- Checks if the id of the active plan of the node matches the input id
- If the ids match, it will return the success status of the matching runningPlan (made a slight change here for better runtime)
- if the ids dont match, it will go through the children looking for the right plan.

Reply:
ah got it